tests: Add memset() unit tests

KhaledEmaraDev opened the pull request:
Closes #1859

anton-bondarev commented on templates/x86/qemu/mods.config:
```diff
@@ -90,6 +90,7 @@ configuration conf {
 	@Runlevel(2) include embox.test.stdlib.qsort_test
 	@Runlevel(2) include embox.test.posix.environ_test
 	@Runlevel(2) include embox.test.posix.getopt_test
+	@Runlevel(2) include embox.test.stdlib.memset_test
```
Rm this please

anton-bondarev commented:
@KhaledEmaraDev Thanks.

but please format at first see code-style ([https://github.com/embox/embox/wiki/Code-Style](https://github.com/embox/embox/wiki/Code-Style) )

KhaledEmaraDev commented:
@anton-bondarev Yeah, sorry about that. I somehow mixed up spaces and tabs, but didn't notice because it looked fine in my editor. I'll set up clang-format. Anyways, removed the test from mods.config and fixed up indentation.

anton-bondarev commented:
@KhaledEmaraDev Please don't rebare your branch now, just push commits under branch head

anton-bondarev commented on src/tests/stdlib/memset.c:
```diff
@@ -0,0 +1,48 @@
+	char str[] = {1, 2, 3, 4, 5};
+	char exp[] = {0, 0, 0, 0, 0};
+
+	memset(str, (int)NULL, sizeof str);
```
Why do you use NULL, It's enough 0 in my mind

anton-bondarev commented:
@KhaledEmaraDev Can you add more tests. for example
memset in the middle of buffer.
It checks that memset don't touch neighbour bytes

KhaledEmaraDev commented:
@anton-bondarev Sorry about the delay. Corrected NULL to 0 and added tests for the middle of the buffer. Tests passed.

anton-bondarev commented:
I've closed PR if you want to contribute these changes please open a new PR

anton-bondarev commented:
I've closed PR if you want to contribute these changes please open a new PR
